addons/misra.py

```diff
@@ -713,7 +713,29 @@ def getForLoopExpressions(forToken):
             lpar.astOperand2.astOperand2.astOperand2]
 
 
-def getForLoopCounterVariables(forToken):
+def get_function_scope(cfg, func):
+    if func:
+        for scope in cfg.scopes:
+            if scope.function == func:
+                return scope
+    return None
+
+
+def is_variable_changed(body_start, body_end, var):
+    """Check if variable is updated between body_start and body_end"""
+    while body_start != body_end:
```

Comment:
nit: in my humble opinion body_start, body_end indicates that we expect that input is { and } tokens. I suggest we rename these to start_token and end_token. I also suggest that in the loop we don't modify the start_token.

Some small refactorings:

    tok = start_token
    while tok != end_token:
        if tok.isAssignmentOp:
             vartok = tok.astOperand1
             while vartok.astOperand1:
                 vartok = vartok1.astOperand1
             if vartok and vartok.variable == var:
                 return True
        tok = tok.next
    return False